src/main/java/delete_post/use_case/DeletePostInteractor.java

```diff
@@ -0,0 +1,37 @@
+package delete_post.use_case;
+
+import delete_post.interface_adapters.DeletePostPresenter;
+import entities.User;
+
+public class DeletePostInteractor implements DeletePostInputBoundary{
+
+    private final DeletePostPresenter presenter;
+    private final DeletePostDsGateway dataAccess;
+
+    public DeletePostInteractor(DeletePostPresenter presenter, DeletePostDsGateway dataAccess){
+        this.presenter = presenter;
+        this.dataAccess = dataAccess;
+    }
+
+    @Override
+    public void delete(DeletePostRequestModel requestModel){
+        if (requestModel.getIsTimer() || requestModel.getIsAdmin() ||
+                requestModel.getUser() == (requestModel.getPostUser())){
+            // delete post
+            DeletePostResponseModel responseModel = new DeletePostResponseModel(requestModel.getPost());
+
+            for (User favouriteId: responseModel.getFavourites()){
+                this.dataAccess.removeFavourite(responseModel.getId(), favouriteId);
```

Comment:
The interactor shouldn't have to use the response model's getters if it has the ability to construct the response model in the first place. For example, responseModel.getID() gives the same result as requestModel.getPost().getID()